packages/context/src/__tests__/unit/binding-sorter.unit.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/context
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {Binding, sortBindingsByGroup} from '../..';
+
+describe('BindingSorter', () => {
+  const orderedGroups = ['log', 'auth'];
+  const groupTagName = 'group';
+  let bindings: Binding<unknown>[];
+  let sortedBindingKeys: string[];
+
+  beforeEach(givenBindings);
+  beforeEach(sortBindings);
+
+  it('sorts by group', () => {
+    assertOrder('logger1', 'logger2', 'auth1', 'auth2');
+  });
+
+  it('sorts by group - unknown group comes before known ones', () => {
+    assertOrder('metrics', 'logger1');
+  });
+
+  it('sorts by group alphabetically without ordered group', () => {
+    assertOrder('metrics', 'rateLimit');
+  });
+
+  it('sorts by binding order without group tags', () => {
+    assertOrder('validator1', 'validator2', 'logger1');
```

Comment:
I'm a bit confused by the name of the test. How is binding order determined in this case? Isn't it just using default group tag of an empty string for `validator1` and `validator2` and alphabetically sorting them followed by the named group tag bindings?

Reply:
Yes, without an explicit `group` tag, the group name is `''` and they are sorted based on that.

Reply:
I added more comments to the tests to make it easier to understand.